design.md

```diff
@@ -2,12 +2,50 @@
 
 ## Who is the target for this design, e.g., are you assuming any knowledge on the part of the language users?
 
+We're still aiming this new language at CS5 students.
+We expect that they understand the picobot model and are thinking of it in
+terms of a state machine.
+Our syntax is also unique - we're not trying to mimic any other language
+(including natural language)
+
 ## Why did you choose this design, i.e., why did you think it would be a good idea for users to express the maze-searching computation using this syntax?
 
+Our goal with our design was to streamline the language. When writing code the
+user should feel like they're specifying the rules for a state machine in the
+most expressive way possible.
```

Comment:
Thinking about the philosophy behind your Piconot language gave me an idea - wouldn't it be cool to be able to write a Picobot program as a finite state machine in JFLAP, and then compile that to run in the Picobot applet (either the one in Scala or JavaScript)? 

In addition, if that language existed would it be an external DSL or an internal one? On one hand, it's written in JFLAP, making it internal to JFLAP. But on the other hand, it couldn't possibly run in JFLAP, so the machine would have to be separately compiled into a set of rules in another language to actually run on the Picobot board, which would make it an external language (because it's not compiled by the host language's compiler)?